api/config/routes.rb

```diff
@@ -133,6 +133,16 @@
       end
     end
 
+    resources :address_book, only: [], param: :address_id do
+      collection do
+        get :mine
+        post :update
+      end
+      member do
+        post :remove
+      end
+    end
```

Comment:
My biggest concern is encapsulation of knowledge and enforcement of constraints. Since only one address (per user) can be default, an edit to one address may cause changes to another (making one default also changes any prior default address). If we treat it in a more traditional Rails-flavored REST-ish way then the client has to know about that and re-request the entire list. We're spreading business logic out into tiers of the app where it doesn't belong. Can we have a brainstorming session in the morning to find the best compromise?

That said, the admin side of things is a good point. However, since this is currently identify users by the auth token, that's the bigger concern.